exchanges/kraken/kraken_websocket.go

```diff
@@ -222,16 +310,24 @@ func (k *Kraken) wsHandleData(respRaw []byte) error {
 						err,
 						respRaw)
 				}
-				if status.ErrorMessage != "" {
-					return fmt.Errorf("%s - err %s",
-						k.Name,
-						status.ErrorMessage)
+				if status.Status == "error" {
+					if !k.Websocket.Match.IncomingWithData(status.RequestID, respRaw) {
+						return fmt.Errorf("can't send ws error message to Matched channel with RequestID: %d, messge: %s",
+							status.RequestID, status.ErrorMessage)
+					}
+					return nil
 				}
+
 				k.Websocket.DataHandler <- &order.Detail{
 					Exchange: k.Name,
 					ID:       status.TransactionID,
 					Status:   order.New,
 				}
+
+				if !k.Websocket.Match.IncomingWithData(status.RequestID, respRaw) {
+					return fmt.Errorf("can't send ws incoming data to Matched channel with RequestID: %d",
+						status.RequestID)
+				}
```

Comment:
With exception to the work you've done with cancelbatchorders, we typically use `k.Websocket.Match.IncomingWithData` to actually return nil.
The reason for this is that if there is a request awaiting a response, we are going to handle it in the requesting function, not here. This is to prevent a double handling of the same data (though its not fully true here, but that's outside of scope).

If you are able to revert the highlighted changes, it will allow this test to work again: https://ci.appveyor.com/project/thrasher-/gocryptotrader-g5pk0/builds/36355457#L254

If you could also remove the test `TestWsCancelOrderJSON` now as it is not needed.

Reply:
optimized functions a bit to allow tests